platforms/jvm/language-java/src/main/java/org/gradle/api/tasks/JavaExec.java

```diff
@@ -176,52 +178,33 @@ private void validateExecutableMatchesToolchain() {
      * {@inheritDoc}
      */
     @Override
-    @ToBeReplacedByLazyProperty
-    public List<String> getAllJvmArgs() {
+    public Provider<List<String>> getAllJvmArgs() {
         return javaExecSpec.getAllJvmArgs();
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public void setAllJvmArgs(List<String> arguments) {
-        javaExecSpec.setAllJvmArgs(arguments);
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public void setAllJvmArgs(Iterable<?> arguments) {
-        javaExecSpec.setAllJvmArgs(arguments);
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    @ToBeReplacedByLazyProperty
+    @ToBeReplacedByLazyProperty(comment = "This property is modified at execution time")
```

Comment:
❓ We will be able to replace these as well with the lenient finalization you just added, right?

Reply:
Yes, we need lenient finalizition merged to upgrade this property